test/test_datasets.py

```diff
@@ -1702,5 +1702,43 @@ def test_classes(self, config):
             self.assertSequenceEqual(dataset.classes, info["classes"])
 
 
+class KittiTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.Kitti
+    FEATURE_TYPES = (PIL.Image.Image, (list, type(None)))  # test split returns None as target
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(split=("train", "test"))
+
+    def inject_fake_data(self, tmpdir, config):
+        kitti_dir = os.path.join(tmpdir, "Kitti", "raw")
+        os.makedirs(kitti_dir)
+
+        split_to_idx = split_to_num_examples = {
+            None: 1,
+            "train": 1,
+            "test": 2,
+        }
+
+        # We need to create all folders regardless of the split in config
+        for split in ("train", "test"):
+            split_idx = split_to_idx[split]
+            num_examples = split_to_num_examples[split]
+
+            datasets_utils.create_image_folder(
+                root=kitti_dir,
+                name=os.path.join(f"{split}ing", "image_2"),
+                file_name_fn=lambda image_idx: f"000{split_idx + image_idx}.png",
```

Comment:
It is not terribly important here, but I have a hard time imagining that the file are named after this scheme in the original dataset. Wouldn't this mean that the second train image (`split_idx==1`, `image_idx==2`) and the first test image (`split_idx==2`, `image_idx==1`) have the same file name?

Reply:
The filenames in the KITTI dataset range from `000000.png` to `007480.png` in the training set and from `000000.png` to `007517.png` in the test set. Here in the test file, for training there will be one image created with `split_idx = 1, image_idx=0` and for test there will be two images with `split_idx=2, image_idx={0, 1}`, so shouldn't have the same file name. Training file name will be `0001.png` and testing file name will be `0002.png` and `0003.png`. Also, even if they were same, it won't matter as they are under different directories `training/image_2` and `testing/image_2`. And yes, I left out a few zeros while naming the files(compared to the actual dataset), which shouldn't really be significant in terms of test results.

Reply:
> Also, even if they were same, it won't matter as they are under different directories `training/image_2` and `testing/image_2`

IMO we should stick to their file naming scheme. True, it doesn't make a difference here, but I stumbled over this during review. Given that they live in separate directories, we can simply use the same file name for every split.



> And yes, I left out a few zeros while naming the files

You can have auto zero padding:

```suggestion
                file_name_fn=lambda image_idx: f"{image_idx:06d}.png",
```

With this the number is padded with zeros in case it has less than 6 digits.